### ISSUE TITLE
Crash when loading a book 

### ISSUE THREAD

**Issue by dolwup** [johnfactotum/foliate#174]
Foliate crashes when loading a book after startup if the "Dark" theme is selected and "Enable Dark Mode" is ticked. This happens if the GTK theme is Adwaita and does not occur while using Adwaita-dark (I only tested it with these two). 

I ran it from terminal and got this output:
```
Gjs-Message: 21:54:36.028: JS WARNING: [resource:///com/github/johnfactotum/Foliate/js/main.js 72]: test for equality (==) mistyped as assignment (=)?
Portal call failed: Failed to start command: Kindprozess »dict« konnte nicht ausgeführt werden (Datei oder Verzeichnis nicht gefunden)
Gjs-Message: 21:54:38.603: JS WARNING: [resource:///com/github/johnfactotum/Foliate/js/main.js 2406]: reference to undefined property "isFullscreen"
```

The first Gjs-Message appeared directly after starting Foliate, the last when Foliate crashed (but only once, I couldn't get it to reappear when running it again).

The german part of the second line translated to english reads: 
"Childprocess »dict« could not be executed (file or directory not found)

I got Foliate from Flathub and it's currently on version 1.5.3

![Peek 2019-09-24 22-28](https://user-images.githubusercontent.com/38048361/65547919-b3313100-df1a-11e9-9920-0bb30d45f3e3.gif)

The same thing happens when I manually select a book using "Open File..."

**Comment by johnfactotum** [comment]
None of the messages has anything to do with a crash. What distro and version are you using?

**Comment by dolwup** [comment]
I'm running antergos (without their own repos, so basically arch)

**Comment by johnfactotum** [comment]
I can't reproduce this issue on Arch Llinux. Perhaps you can try [building a flatpak manually](https://github.com/johnfactotum/foliate#building-flatpaks-manually) and see if it happens with previous versions, or the current git master version.

**Comment by dolwup** [comment]
I built it from the current git master and I built the flatpak using flatpak-builder and the result was that the problem only occured in the flatpak version. The github build worked flawlessly

**Comment by johnfactotum** [comment]
Mysterious crash only in flatpak? And I still can't reproduce the problem. This is #88 all over again...

**Comment by swltk** [comment]
Hello, not sure if this helps, I get the same error messages that @dolwup listed, but I get the additional line of:

`file:///app/share/com.github.johnfactotum.Foliate/assets/epub.js:15617:19: CONSOLE ERROR [object Object]`

Using the latest Flathub version on Debian Sid (currently on Flatpak 1.4.3).

**Comment by mgh520** [comment]
Crashes for me too when opening epub on Centos 8


```
flatpak run com.github.johnfactotum.Foliate
Portal call failed: Failed to start command: Failed to execute child process “dict” (No such file or directory)

(com.github.johnfactotum.Foliate:2): Gdk-WARNING **: 21:52:27.206: Failed to read portal settings: GDBus.Error:org.freedesktop.DBus.Error.UnknownMethod: No such interface 'org.freedesktop.portal.Settings' on object at path /org/freedesktop/portal/desktop
Gjs-Message: 21:52:27.610: JS WARNING: [resource:///com/github/johnfactotum/Foliate/js/window.js 1119]: reference to undefined property "_loading"
Gjs-Message: 21:52:33.237: JS WARNING: [resource:///com/github/johnfactotum/Foliate/js/tts.js 33]: reference to undefined property "_epub"

(WebKitWebProcess:40): Gdk-WARNING **: 21:52:33.324: Failed to read portal settings: GDBus.Error:org.freedesktop.DBus.Error.UnknownMethod: No such interface 'org.freedesktop.portal.Settings' on object at path /org/freedesktop/portal/desktop
file:///app/share/com.github.johnfactotum.Foliate/assets/epub-viewer.js:352:22: CONSOLE ERROR TypeError: undefined is not an object (evaluating 'cfiOne.path.steps')
Gdk-Message: 21:52:34.665: Error flushing display: Protocol error

```

**Comment by johnfactotum** [comment]
@mgh520 the `Gdk-WARNING`s and `Gdk-Message` probably indicate something wrong in your Flatpak setup, although I'm not sure if they have anything to do with the crash.

**Comment by johnfactotum** [comment]
Closing this now since I've not been able to reproduce the problem and Foliate has since been rewritten. If this still happens with the latest version, please open a new issue.
